fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraVersioning.java

```diff
@@ -191,4 +195,14 @@ protected FedoraResource resource() {
     protected Session session() {
         return session;
     }
+
+    /**
+     * Determines if a label contains valid characters
+     * @param label label
+     * @return whether label contains valid characters
+     */
+    public boolean invalidLabel(final String label) {
```

Comment:
In looking at this closer, the label validation should be enforced even if the user does not make the request via the REST API. Therefore, you should move your updates deeper into the application, i.e. https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/VersionServiceImpl.java#L162